index.js

```diff
@@ -0,0 +1,131 @@
+const express = require('express');
+const cors = require('cors');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session);
+const jwt = require('jsonwebtoken');
+const db = require('./database/dbConfig.js');
+const server = express();
+const port = 9000;
+
+const sessionConfig = {
+  secret: 'nobody-tosses.a%dwarf.!',
+  name: 'monkey',
+  httpOnly: true,
+  resave: false,
+  saveUninitialized: false,
+  cookie: {
+    secure: false,
+    maxAge: 1000 * 60 * 3,
+  },
+  store: new KnexSessionStore({
+    tablename: 'sessions',
+    sidfieldname: 'sid',
+    knex: db,
+    createtable: true,
+    clearInterval: 1000 * 60 * 60,
+  })
+};
+
+server.use(session(sessionConfig));
+server.use(express.json());
+server.use(cors());
+
+
+// ======== SERVER RUNNING ======== //
+server.get('/', (req, res) => {
+  res.send('<h1>Its Alive!<h1>');
+});
+
+
+// ========= CREATES USER ========= //
+server.post('/register', (req, res) => {
+  const credentials = req.body;
+  const hash = bcrypt.hashSync(credentials.password, 10);
+  credentials.password = hash;
+
+  db('users')
+    .insert(credentials)
+    .then(ids => {
+      const id = ids[0];
+      res.status(201).json({ newUserId: id });
+    })
+    .catch(err => {
+      res.status(500).json(err);
+    });
+});
+
+
+// ======= GENERATES TOKEN ======= //
+const jwtSecret = 'nobody tosses a dwarf!';
+
+function generateToken(user) {
+  const jwtPayload = {
+    ...user,
+    hello: 'FSW13',
+    role: 'admin'
+  };
+  const jwtOptions = {
+    expiresIn: '5m',
+  };
+  return jwt.sign(jwtPayload, jwtSecret, jwtOptions);
+}
+
+
+// ========== LOGIN ========= //
+server.post('/login', (req, res) => {
+  const creds = req.body;
```

Comment:
It would be best to specify that you want only the username and password from the request body.